pkg/execution/state/redis_state/lua/cancel.lua

```diff
@@ -11,9 +11,7 @@ Output:
 local metadataKey = KEYS[1]
 
 local value = tonumber(redis.call("HGET", metadataKey, "status"))
-
--- If run has ended (completed, failed, etc.)
-if value ~= 0 and value ~= 5 then
+if value ~= 0 then
```

Comment:
If we're adding queued and running, should we check for both statuses here?

Reply:
Update:  it looks like this won't work for scheduled statuses?

Reply:
The state store doesn't differentiate between scheduled and running statuses. Right now we have to either:
- Have a scheduled status but never transition to running (the behavior in the `main` branch right now)
- Have a running status that's never preceded by the scheduled status (the behavior before the PR I'm reverting)